src/main/java/com/nutomic/syncthingandroid/fragments/SelectFolderFragment.java

```diff
@@ -0,0 +1,75 @@
+package com.nutomic.syncthingandroid.fragments;
+
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+
+public class SelectFolderFragment extends FoldersFragment {
+
+    private static final String TAG = "SelectFolderFragment";
+
+    @Override
+    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+        for (Uri uri : files) {
+            File sourceFile = new File(getRealPathFromURI(uri));
+            String path = mAdapter.getItem(i).path;
+            File destFile = new File(path, sourceFile.getName());
+            try {
+                copy(sourceFile, destFile);
+            }
+            catch (IOException e) {
+                Log.e(TAG, "Failed copying file", e);
+            }
+        }
+
+        getActivity().finish();
+    }
+
+    public void setFiles(ArrayList<Uri> files) {
```

Comment:
The files are not available until `ShareToFolderActivity`'s `onCreate` method is called. I'm not sure how to pass `SelectFolderFragment`'s constructor different parameters since it's declared in XML. What's the best way to handle this? Should I call the constructor programatically and infalte the XML?